src/app/common/components/MustGatherModal.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { Button, TextContent, Modal } from '@patternfly/react-core';
+import { MustGatherContext, AppContext, mustGatherObjType } from '@app/common/context';
+import { useMustGatherMutation } from '@app/queries';
+
+export const MustGatherModal: React.FunctionComponent = () => {
+  const mustGatherContext = React.useContext(MustGatherContext);
+  const { pushNotification } = React.useContext(AppContext);
+
+  const handleMustGatherSuccess = () => {
+    if (mustGatherContext.activeMustGather) {
+      pushNotification({
+        title: `Started must gather for ${mustGatherContext.activeMustGather.customName}`,
+        message: '',
+        key: mustGatherContext.activeMustGather.customName,
+        variant: 'info',
+        actionClose: true,
+        timeout: 4000,
+      });
+    }
+  };
+
+  const handleMustGatherError = () => {
+    if (mustGatherContext.activeMustGather) {
+      pushNotification({
+        title: `Could not run must gather for ${mustGatherContext.activeMustGather.customName}`,
+        message: '',
+        key: mustGatherContext.activeMustGather.customName,
+        variant: 'danger',
+        actionClose: true,
+      });
+    }
+  };
+
+  const registerMustGather = useMustGatherMutation(
+    'must-gather',
+    handleMustGatherSuccess,
+    handleMustGatherError
+  );
+
+  const handleMustGatherRequest = ({ customName, type }: mustGatherObjType) => {
```

Comment:
Nitpick, but it feels weird to have `mustGatherObjType` not be capitalized. At first glance it seems like a value and not a type (despite the name and syntax... take it or leave it lol)